### PR TITLE
`resource/alicloud_log_alert`: add auto_annotation and power_sql_mode in alert config; update log sdk. 

### PR DESCRIPTION
**Pull request by panawala**
* update log sdk. 
*  add auto_annotation and power_sql_mode in alert config.

### CODE REVIEW

**Comment by panawala** [comment]
$TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudLogAlert -timeout=300m
=== RUN   TestAccAlicloudLogAlertResourceDataSource
--- PASS: TestAccAlicloudLogAlertResourceDataSource (15.52s)
=== RUN   TestAccAlicloudLogAlert_basic
--- PASS: TestAccAlicloudLogAlert_basic (94.24s)
=== RUN   TestAccAlicloudLogAlert_basic_new
--- PASS: TestAccAlicloudLogAlert_basic_new (96.87s)
PASS
ok      github.com/aliyun/terraform-provider-alicloud/alicloud  (cached)

File: alicloud/resource_alicloud_log_alert.go [hunk 1]
```diff
@@ -248,7 +257,7 @@ func resourceAlicloudLogAlert() *schema.Resource {
 						"type": {
 							Type:         schema.TypeString,
 							Required:     true,
-							ValidateFunc: validation.StringInSlice([]string{"cross_join", "inner_join", "left_join", "right_join", "full_join", "left_exclude", "right_exclude", "concat", "no_join"}, true),
+							ValidateFunc: validation.StringInSlice([]string{"cross_join", "inner_join", "left_join", "right_join", "full_join", "left_exclude", "right_exclude", "concat", "no_join"}, false),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_log_alert.go:260]: 这个为啥要改

File: alicloud/resource_alicloud_log_alert.go [hunk 2]
```diff
@@ -793,6 +805,11 @@ func createAlert2Config(d *schema.ResourceData) *sls.AlertConfiguration {
 	if !ok {
 		configType = "default"
 	}
+	autoAnnotation := false
+	_, ok = d.GetOk("auto_annotation")
+	if ok {
+		autoAnnotation = d.Get("auto_annotation").(bool)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_log_alert.go:811]: 这个地方直接用 
if v, ok := d.GetOk("auto_annotation"); ok  {
  autoAnnotation = v.(bool)
}

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/resource_alicloud_log_alert.go (hunk 1 above)
Comment (panawala) [reply alicloud/resource_alicloud_log_alert.go:260]: 这里大小写是要区分的，之前应该是写错了

**Comment by panawala** [comment]
$TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudLogAlert -timeout=300m
=== RUN   TestAccAlicloudLogAlertResourceDataSource
--- PASS: TestAccAlicloudLogAlertResourceDataSource (15.57s)
=== RUN   TestAccAlicloudLogAlert_basic
--- PASS: TestAccAlicloudLogAlert_basic (93.95s)
=== RUN   TestAccAlicloudLogAlert_basic_new
--- PASS: TestAccAlicloudLogAlert_basic_new (98.50s)
PASS
ok      github.com/aliyun/terraform-provider-alicloud/alicloud  208.407s

File: alicloud/resource_alicloud_log_alert.go (hunk 1 above)
Comment (panawala) [reply alicloud/resource_alicloud_log_alert.go:260]: 这个是大小写敏感，之前应该写的有问题

**Pull request merged**